Userland/Utilities/tail.cpp

```diff
@@ -77,19 +52,62 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     TRY(Core::System::pledge("stdio rpath"));
 
     bool follow = false;
-    int line_count = DEFAULT_LINE_COUNT;
-    char const* file = nullptr;
+    size_t line_count = DEFAULT_LINE_COUNT;
+    StringView file;
 
     Core::ArgsParser args_parser;
     args_parser.set_general_help("Print the end ('tail') of a file.");
     args_parser.add_option(follow, "Output data as it is written to the file", "follow", 'f');
     args_parser.add_option(line_count, "Fetch the specified number of lines", "lines", 'n', "number");
-    args_parser.add_positional_argument(file, "File path", "file");
+    args_parser.add_positional_argument(file, "File path", "file", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
-    auto f = TRY(Core::File::open(file, Core::OpenMode::ReadOnly));
-    TRY(Core::System::pledge("stdio"));
+    auto f = TRY(Core::Stream::File::open_file_or_standard_stream(file, Core::Stream::OpenMode::Read));
+    if (!follow)
+        TRY(Core::System::pledge("stdio"));
 
-    auto pos = find_seek_pos(*f, line_count);
-    return tail_from_pos(*f, pos, follow);
+    // Since we can't seek the standard input, we read
+    // the entire file and split on new lines.
+    if (f->is_standard_input()) {
+        do {
+            // FIXME: f->read_all() does not block anymore after
+            // sending EOF (^D), despite f->is_open() returning true.
+            auto buffer = TRY(f->read_all(PAGE_SIZE));
+            auto lines = StringView(buffer).split_view("\n"sv, SplitBehavior::KeepEmpty | SplitBehavior::KeepTrailingSeparator);
+            long i = 0;
```

Comment:
There is no need for this to be a `long`, `size_t` should be completely fine for this.